### PR TITLE
Apply auto close braces in code block

### PR DESCRIPTION
**Pull request by LavaCxx**
---

- [x] I understand that contributing to this repository will require me to agree with the [CLA](https://github.com/anyproto/.github/blob/main/docs/CLA.md)

---

### Description
To provide some convenience for directly writing code in code blocks.
Also provided some additional support for full-width symbols, such as `【「(“‘` etc.

### What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI

### Related Tickets & Documents
#395 

### Mobile & Desktop Screenshots/Recordings

demo

### Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [x] 🙋 no, because I need help

### Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 [tech-docs](https://github.com/anyproto/tech-docs)
- [x] 🙅 no documentation needed

### CODE REVIEW

File: src/ts/component/block/text.tsx [hunk 1]
```diff
@@ -643,7 +643,7 @@ const BlockText = observer(class BlockText extends React.Component<Props> {
 		e.persist();
 
 		const { onKeyDown, rootId, block, isInsideTable } = this.props;
-		const { id } = block;
+		const { id,content:{text} } = block;
```
Comment (ra3orblade) [inline src/ts/component/block/text.tsx:646]: I do not see where it's used and we do not write this way, `const { text } = block.content;` for example.

File: src/ts/component/block/text.tsx [hunk 2]
```diff
@@ -702,12 +702,17 @@ const BlockText = observer(class BlockText extends React.Component<Props> {
 				saveKeys.push({ key: `arrowright, arrowdown` });
 			};
 		};
-
-		const twineOpen = [ '[', '{', '\'', '\"', '(' ];
+		[];
```
Comment (ra3orblade) [inline src/ts/component/block/text.tsx:705]: Useless braces?

File: src/ts/component/block/text.tsx [hunk 3]
```diff
@@ -702,12 +702,17 @@ const BlockText = observer(class BlockText extends React.Component<Props> {
...
+		const twineOpen = [ '[', '{', '\'', '\"', '`', '(', '【', '「', '（', '“', '‘' ];
```
Comment (ra3orblade) [inline src/ts/component/block/text.tsx:706]: This needs refactoring, we can leave only one map and use it's keys as opening symbols.

File: src/ts/component/block/text.tsx [hunk 4]
```diff
@@ -859,7 +863,7 @@ const BlockText = observer(class BlockText extends React.Component<Props> {
 				focus.set(block.id, { from: range.from + l, to: range.to + l });
 				focus.apply();
 			});
-
+			{}
```
Comment (ra3orblade) [inline src/ts/component/block/text.tsx:866]: Unused braces?

**Review by ra3orblade — CHANGES_REQUESTED** [review]

File: src/ts/component/block/text.tsx (hunk 1 above)
Comment (LavaCxx) [reply src/ts/component/block/text.tsx:646]: Sorry, I forgot to delete some test codes. The original intention was to try automatic indentation.

File: src/ts/component/block/text.tsx (hunk 4 above)
Comment (LavaCxx) [reply src/ts/component/block/text.tsx]: I tested how vscode handles indentation on the same page, but I forgot to delete it, which was a mistake that should not have happened.

File: src/ts/component/block/text.tsx [hunk 5]
```diff
@@ -843,13 +850,12 @@ const BlockText = observer(class BlockText extends React.Component<Props> {
 			e.preventDefault();
 			this.onSmile();
 		});
-
-		if (range && (range.from != range.to) && twineOpen.includes(key)) {
+		if (range && ((range.from != range.to) || block.isTextCode()) && twinePairs[key]) {
```
Comment (ra3orblade) [inline src/ts/component/block/text.tsx:853]: This will work with a closing symbol as well, `Object.keys(twinePairs).includes(key)` I meant

**Review by ra3orblade — CHANGES_REQUESTED** [review]

File: src/ts/component/block/text.tsx (hunk 5 above)
Comment (LavaCxx) [reply src/ts/component/block/text.tsx]: How about `Reflect.has(twinePairs,key)`

Comment (ra3orblade) [reply src/ts/component/block/text.tsx]: Ok, never used it though)

**Review by ra3orblade — APPROVED** [review]

**Pull request merged**